dask/utils.py

```diff
@@ -1032,3 +1035,29 @@ def __str__(self):
         return "<%s: %s>" % (self.__class__.__name__, self.token)
 
     __repr__ = __str__
+
+
+def effective_get(get=None, df=None):
+    """Get the effective get method used in a given situation"""
+    df_get = df._default_get if df else None
+    return get or _globals.get('get') or df_get
```

Comment:
We might want to remove `effective_get` from utils.py and instead split this into two methods (one to get the `get` and another to call `_optimize` and `get`. I think it'll look better in there. 

https://github.com/dask/dask/blob/master/dask/base.py#L83